core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnSubstringAfter.java

```diff
@@ -0,0 +1,90 @@
+import gov.nist.secauto.metaschema.core.metapath.function.FunctionUtils;
+import gov.nist.secauto.metaschema.core.metapath.function.IArgument;
+import gov.nist.secauto.metaschema.core.metapath.function.IFunction;
+import gov.nist.secauto.metaschema.core.metapath.item.IItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IDecimalItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IStringItem;
+import gov.nist.secauto.metaschema.core.util.ObjectUtils;
+
+import java.util.List;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Implements <a href=
+ * "https://www.w3.org/TR/xpath-functions-31/#func-substring-after">fn:substring-after</a>.
+ */
+public final class FnSubstringAfter {
+  private static final String NAME = "substring-after";
+  @NonNull
+  static final IFunction SIGNATURE_TWO_ARG = IFunction.builder()
+      .name(NAME)
+      .namespace(MetapathConstants.NS_METAPATH_FUNCTIONS)
+      .deterministic()
+      .contextIndependent()
+      .focusIndependent()
+      .argument(IArgument.builder()
+          .name("arg1")
+          .type(IStringItem.class)
+          .zeroOrOne()
+          .build())
+      .argument(IArgument.builder()
+          .name("arg2")
+          .type(IStringItem.class)
+          .zeroOrOne()
+          .build())
+      .returnType(IStringItem.class)
+      .returnOne()
+      .functionHandler(FnSubstringAfter::executeTwoArg)
+      .build();
+
+  private FnSubstringAfter() {
+    // disable construction
+  }
+
+  @SuppressWarnings({ "unused", "PMD.OnlyOneReturn" })
+  @NonNull
+  private static ISequence<IStringItem> executeTwoArg(
+      @NonNull IFunction function,
+      @NonNull List<ISequence<?>> arguments,
+      @NonNull DynamicContext dynamicContext,
+      IItem focus) {
+	
+	// From the XPath 3.1 specification:
+	// If the value of $arg1 or $arg2 is the empty sequence, or contains only
+	// ignorable collation units, it is interpreted as the zero-length string.
+    IStringItem arg1 = arguments.get(0) == ISequence.empty() ? IStringItem.valueOf("") : FunctionUtils.asTypeOrNull(arguments.get(0).getFirstItem(true));
+    IStringItem arg2 = arguments.get(1) == ISequence.empty() ? IStringItem.valueOf("") : FunctionUtils.asTypeOrNull(arguments.get(1).getFirstItem(true));
+    
+    return ISequence.of(IStringItem.valueOf(fnSubstringAfter(arg1.asString(), arg2.asString())));
+  }
+
+  /**
+   * An implementation of XPath 3.1 <a href=
+   * "https://www.w3.org/TR/xpath-functions-31/#func-substring-after">fn:substring-after</a>.
+   *
+   * @param arg1
+   *          the source string to get a substring from
+   * @param arg2
+   *          the substring to match and find the substring to return after the match
+   * @return the substring
+   */
+  @NonNull
+  public static String fnSubstringAfter(
+      @NonNull String arg1,
+      @NonNull String arg2) {
+    return StringUtils.substringAfterLast(arg1, arg2);
```

Comment:
I am not sure that [StringUtils.html#substringAfterLast\(java.lang.String,java.lang.String\)](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringAfterLast\(java.lang.String,java.lang.String\)) is the right method to use, since it is not the first minimal match as [defined in the specification](https://www.w3.org/TR/xpath-functions-31/#func-substring-after). It might be better to use [StringUtils.html#substringAfter\(java.lang.String,java.lang.String\)](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringAfter\(java.lang.String,java.lang.String\))

Reply:
Ok I’m fine with that. Let me switch it out.

```suggestion
    return StringUtils.substringAfter(arg1, arg2);
```